src/hooks/common/authorization/useConfidentialsAuthorization.ts

```diff
@@ -0,0 +1,40 @@
+import { useAppState } from '@/contexts/appState'
+import { useMemo, useState } from 'react'
+
+export function useConfidentialsAuthorization(): boolean {
+  const [state] = useAppState()
+  const {
+    connection: { account },
+    manager: { voucherManager, confidentialManager },
+    authorization: { confidentials },
+  } = state
+  const [accessConfidential, setAccessConfidential] = useState(confidentials)
+
+  useMemo(async () => {
```

Comment:
async fetching shouldn't be performed inside `useMemo` for performance reasons (you are blocking the interface until the call is resolved, so in the worst case scenario it can take minutes.

Use `useState` + `useEffect` to handle it, and render an spinner while the data is loading. We have a hook called `useRequest` in core that can handle this scenarios